src/ucp/api/ucp.h

```diff
@@ -2140,19 +2127,12 @@ ucs_status_t ucp_request_test(void *request, ucp_tag_recv_info_t *info);
  *
  * @param [in]  request      Non-blocking request to release.
  *
- * This routine marks and potentially releases back to the library the
- * non-blocking request. If the request is already completed or
- * canceled state it is released and the resources associated with the request are
- * returned back to the library.  If the request in any other stated it is
- * marked as a "ready to release" and it will be released once it enters
- * completed and canceled states.
- *
- * @todo I think this release semantics is a bit confusing. I would suggest to
- * remove the "marked" ready to release. Instead, user has to call the release
- * explicitly once it is completed or released and we should return an error if
- * the request is not in one of these states.
+ * This routine releases the non-blocking request back to the library, regardless
```

Comment:
There is a doxygen/latex problem.  These changes now result in an unwanted page break in ucx.pdf before the parameter table for ucp_request_free().  This is with doxygen 1.8.6 on ubuntu 14.04.  The only way I know offhand to circumvent this is to put the param information after the function description, rather than before.  We currently do it both ways in the document, so, it would not be inconsistent (although not necessarily desirable) with what's already done.  @yosefe do you know of a better way to fix this?